ngraph/core/reference/include/ngraph/runtime/reference/floor_mod.hpp

```diff
@@ -26,7 +26,10 @@ namespace ngraph
             {
                 autobroadcast_binop(
                     arg0, arg1, out, arg0_shape, arg1_shape, broadcast_spec, [](T x, T y) -> T {
-                        return x - y * std::floor(x / y);
+                        // Cast to double is needed for integer input,
+                        // otherwise std::floor will act like std::trunc
+                        const double divisor = static_cast<double>(y);
+                        return x - y * std::floor(x / divisor);
```

Comment:
Can you add backend or evaluate test to cover this change?

Reply:
Done